pkg/cnb/create_builder.go

```diff
@@ -110,6 +128,20 @@ func (r *RemoteBuilderCreator) CreateBuilder(ctx context.Context, builderKeychai
 	return builder, nil
 }
 
+func ResolveBuilderRef(ref, builderKind, builderName string) (string, error) {
+	parsedRef, err := name.ParseReference(ref)
+	if err != nil {
+		return "", err
+	}
+
+	// this happens if there is no tag
+	if parsedRef.Identifier() == "latest" {
+		return parsedRef.Context().Tag(fmt.Sprintf("%s-%s", builderKind, builderName)).Name(), nil
+	}
+
+	return parsedRef.Name(), nil
+}
+
```

Comment:
nit: this function should be tested

Reply:
ended up just moving the implementations into each reconciler. I realized I needed to include the namespace in the builder tag so the implementation is slightly different.